ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,260 @@
+# Copyright OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Cloud Trace Span Exporter for OpenTelemetry."""
+
+import logging
+from typing import Any, Dict, List, Sequence, Tuple
+
+from google.cloud.trace_v2 import TraceServiceClient
+from google.cloud.trace_v2.proto.trace_pb2 import AttributeValue
+from google.cloud.trace_v2.proto.trace_pb2 import Span as ProtoSpan
+from google.cloud.trace_v2.proto.trace_pb2 import TruncatableString
+from google.rpc.status_pb2 import Status
+
+import opentelemetry.trace as trace_api
+from opentelemetry.sdk.trace import Event
+from opentelemetry.sdk.trace.export import Span, SpanExporter, SpanExportResult
+from opentelemetry.util import types
+
+logger = logging.getLogger(__name__)
+
+# Max length is 128 bytes for a truncatable string.
+MAX_LENGTH = 128
+
+
+class CloudTraceSpanExporter(SpanExporter):
+    """Cloud Trace span exporter for OpenTelemetry.
+
+    Args:
+        project_id: ID of the cloud project that will receive the traces.
+        client: Cloud Trace client. If not given, will be taken from gcloud
+            default credentials
+    """
+
+    def __init__(
+        self, project_id, client=None,
+    ):
+        self.client = client or TraceServiceClient()
+        self.project_id = project_id
+
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        """Export the spans to Cloud Trace.
+
+        See: https://cloud.google.com/trace/docs/reference/v2/rest/v2/projects.traces/batchWrite
+
+        Args:
+            spans: Tuple of spans to export
+        """
+        cloud_trace_spans = []
+        for span in self.translate_to_cloud_trace(spans):
+            try:
+                cloud_trace_spans.append(self.client.create_span(**span))
+            # pylint: disable=broad-except
+            except Exception as ex:
+                logger.warning("Error %s when creating span %s", ex, span)
+
+        try:
+            self.client.batch_write_spans(
+                "projects/{}".format(self.project_id), cloud_trace_spans,
+            )
+        # pylint: disable=broad-except
+        except Exception as ex:
+            logger.warning("Error while writing to Cloud Trace: %s", ex)
+            return SpanExportResult.FAILURE
```

Comment:
Not specific to this PR but should SpanExportResult be able to encapsulate other error information like the original exception? If all exporters try to catch and swallow every exception globally and then return just one value to represent failure, we might be losing out on valuable information. Perhaps processor/exporter should be able to return the original python exceptions or error messages+stack traces along with a failure status. This information can be used by the caller to show helpful error messages to users by utilizing a single global error handler instead of relying on every processor to do the right thing by using the correct logger. In general, I feel error handling should propagate back up and be taken care of preferably in a single place. Especially if we are catching all possible exceptions.

Reply:
I agree with you, but I would prefer to handle that in a seperate PR :)

Reply:
For sure, I'm not suggesting you change it in this PR but wanted to bring this up for the Python group in general.

Reply:
I agree, although in this case hopefully the logger.warning helps.